presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestAggregations.java

```diff
@@ -660,12 +660,71 @@ public void test15WayGroupBy()
     @Test
     public void testApproximateCountDistinct()
     {
+        // test date
+        assertQuery("SELECT approx_distinct(orderdate) FROM orders", "SELECT 2443");
+        assertQuery("SELECT approx_distinct(orderdate, 0.023) FROM orders", "SELECT 2443");
+
+        // test timestamp
+        assertQuery("SELECT approx_distinct(CAST(orderdate AS TIMESTAMP)) FROM orders", "SELECT 2384");
+        assertQuery("SELECT approx_distinct(CAST(orderdate AS TIMESTAMP), 0.023) FROM orders", "SELECT 2384");
+
+        // test timestamp with time zone
+        assertQuery("SELECT approx_distinct(CAST((orderdate) AS TIMESTAMP WITH TIME ZONE)) FROM orders", "SELECT 2384");
+        assertQuery("SELECT approx_distinct(CAST((orderdate) AS TIMESTAMP WITH TIME ZONE), 0.023) FROM orders", "SELECT 2384");
+
+        // test day to second interval
+        assertQuery("SELECT approx_distinct(\"$literal$interval day to second\"(custkey)) FROM orders", "SELECT 990");
+        assertQuery("SELECT approx_distinct(\"$literal$interval day to second\"(custkey), 0.023) FROM orders", "SELECT 990");
+
+        // test year to month interval
+        assertQuery("SELECT approx_distinct(\"$literal$interval year to month\"(custkey)) FROM orders", "SELECT 990");
+        assertQuery("SELECT approx_distinct(\"$literal$interval year to month\"(custkey), 0.023) FROM orders", "SELECT 990");
+
+        // test time
+        assertQuery("SELECT approx_distinct(\"$literal$time\"(custkey)) FROM orders", "SELECT 990");
```

Comment:
`CAST(from_unixtime(custkey) AS TIME)`. Implicit cast of custkey to double and than back to long changes the estimate to `993` though